src/devices/BrickPi3/README.md

```diff
@@ -0,0 +1,311 @@
+
+You will need as well Lego NXT or Lego EV3 sensors and or motors to fully enjoy this device. See [Lego Mindstorms](https://www.lego.com/en-us/mindstorms) for more information.
+
+- [Device family](./README.md#device-family)
+- [BrickPi3 requirements](./README.md#make-sure-you-have-a-brickpi3)
+- [Know limitations](./README.md#known-limitations)
+- [Using the driver](./README.md#how-to-use-the-driver)
+  - [Accessing BrickPi3 information](./README.md#accessing-brickpi3-information)
+  - [Using sensors](./README.md#using-sensors)
+  - [Using motors](./README.md#using-motors)
+    - [Making a motor moving depending on the position of another motor](./README.md#making-a-motor-moving-depending-on-the-position-of-another-motor)
+    - [Setting power to motors](./README.md#setting-power-to-motors)
+    - [Reading encoders](./README.md#reading-encoders)
+    - [Setting motor speed with degree per seconds](./README.md#setting-motor-speed-with-degree-per-seconds)
+- [Using high level classes](./README.md#how-to-use-the-high-level-classes)
+  - [Using the Sensor classes](./README.md#using-the-sensor-classes)
+  - [Using Motors](./README.md#using-motors)
+- [Using GrovePi port](./README.md#using-grovepi-port)
+- [Tests](./README.md#tests)
+
+## Device family
+
+The device supported is the BrickPi3 [Dexter Industries](https://www.dexterindustries.com/brickpi/). It does implement all the features offered to pilot motors, get values from sensor. The first version of BrickPi is not supported.
+
+![image how it works](BrickPi3-How-it-Works.jpg)
+
+More information on [Dexter Industries website](https://www.dexterindustries.com/BrickPi/brickpi3-technical-design-details/).
+
+## Make sure you have a BrickPi3
+
+There are multiple versions of BrickPi, this code is only working for BrickPi3. The main difference between the 2 BrickPi is the switch available close to the battery alimentation. If you have a switch, you have a BrickPi3.
+If you have a previous version, you can find the code on for Windows 10 IoT Core on [Laurent Ellerbach's GitHub](https://github.com/ellerbach/BrickPi). So far, there is no port planned from this Windows 10 IoT Core version.
+
+## Known limitations
+
+This version include a pure driver so you can directly access to the raw results. High level classes has been created and most of the NXT and EV3 sensors has been tested. Please open a GitHub issue if you have any issue with any sensor. 
+
+For I2C sensors, the code has been tested very little. Color sensors may return incorrect data, you need to allow time when measuring and not calling a refresh of the sensor too often. This is a limitation of the sensor itself.
+
+## How to use the driver
+
+The main [BrickPi3.samples](./samples) contains a series of test showing how to use every elements of the driver.
+
+Create a ```Brick``` class.
+
+```csharp
+Brick brick = new Brick();
```

Comment:
In here where are you setting the SPIConnection settings? and also what about chipSelect?

Reply:
They are in the constructor. As the shield is designed for Raspberry Pi, it is supposed to be fixed. That said, if needed, I can add in the constructor. So it will allow to use the shield with other boards easilly